tests/test_dsp.py

```diff
@@ -212,6 +213,66 @@ def test_regu_inversion(impulse):
     npt.assert_allclose(res.freq[:, -1], [0.25])
 
 
+def test_time_shift():
+    sampling_rate = 100
+    delay = 2
+    n_samples = 10
+    test_signal = impulse(n_samples, delay=delay, sampling_rate=sampling_rate)
+
+    # shift to the right
+    shift_samples = 2
+    shifted = dsp.time_shift(test_signal, shift_samples, unit='samples')
+    ref = impulse(
+        n_samples, delay=delay+shift_samples, sampling_rate=sampling_rate)
+
+    npt.assert_allclose(shifted.time, ref.time)
```

Comment:
Can you add a line for testing if the input remained unchanged for explicity? It's implicitly tested already :)

Reply:
:+1: